packages/server/graphql/private/mutations/removeAuthIdentity.ts

```diff
@@ -0,0 +1,80 @@
+import base64url from 'base64url'
+import bcrypt from 'bcrypt'
+import crypto from 'crypto'
+import {AuthIdentityTypeEnum, Security} from 'parabol-client/types/constEnums'
+import {r} from 'rethinkdb-ts'
+import util from 'util'
+import AuthIdentityLocal from '../../../database/types/AuthIdentityLocal'
+import PasswordResetRequest from '../../../database/types/PasswordResetRequest'
+import getMailManager from '../../../email/getMailManager'
+import resetPasswordEmailCreator from '../../../email/resetPasswordEmailCreator'
+import generateRandomString from '../../../generateRandomString'
+import getUsersByDomain from '../../../postgres/queries/getUsersByDomain'
+import updateUser from '../../../postgres/queries/updateUser'
+import {MutationResolvers} from '../../private/resolverTypes'
+
+const randomBytes = util.promisify(crypto.randomBytes)
+
+const removeAuthIdentity: MutationResolvers['removeAuthIdentity'] = async (
+  _source,
+  {domain, identityType, addLocal},
+  {ip}
+) => {
+  // VALIDATION
+  const users = await getUsersByDomain(domain)
+  if (!users.length) {
+    return {error: {message: 'No user emails match that domain'}}
+  }
+
+  // RESOLUTION
+  const updateUsersPromises = users.map(async (user) => {
+    const {id: userId, identities} = user
+    const filteredIdentities = identities.filter((identity) => identity.type !== identityType)
+    if (!filteredIdentities.length || addLocal) {
+      const identityId = `${userId}:${AuthIdentityTypeEnum.LOCAL}`
```

Comment:
The `identityId` will look a lil strange as it includes both the google auth & local strings, e.g. `google-oauth2|e2N8F8q7II:LOCAL`. I think that's OK as long as the `id` is unique as we're using the `type` field rather than the `id` to know which type it is.